hles/transformation/src/main/scala/org/broadinstitute/monster/dap/RawRecord.scala

```diff
@@ -50,7 +52,24 @@ case class RawRecord(id: Long, fields: Map[String, Array[String]]) {
   def getOptionalBoolean(field: String): Option[Boolean] = getOptional(field).map(_ == "1")
 
   /** Get the singleton value for an attribute in this record, parsed as a long. */
-  def getOptionalNumber(field: String): Option[Long] = getOptional(field).map(_.toLong)
+  def getOptionalNumber(field: String, truncateDecimals: Boolean = true): Option[Long] =
+    getOptional(field).map(value => {
+      // we can presume the string represents a positive integer if all of its characters are decimal digits
+      if (truncateDecimals && !value.forall(Character.isDigit)) {
+        val truncatedValue: Long = value.toFloat.toLong
+
+        // don't log this error message until after we've successfully converted the string to a long.
+        // this avoids us logging this message erroneously if we were unable to parse the value,
+        // e.g. because it was garbled nonsense
+        TruncatedDecimalError(
```

Comment:
OK, I'm a dope and didn't grok the full intent of this codepath (try _and_ log in the event of a bogus decimal value). In that case, I think the EAFP approach is the best way to go rather than the string parsing here, sorry about the confusion.